config/confighttp/confighttp.go

```diff
@@ -53,9 +55,13 @@ type HTTPClientSettings struct {
 	// Timeout parameter configures `http.Client.Timeout`.
 	Timeout time.Duration `mapstructure:"timeout"`
 
+	// Deprecated: [v0.67.0] Use OpaqueHeaders instead.
+	Headers map[string]string `mapstructure:"-"`
+
 	// Additional headers attached to each HTTP request sent by the client.
 	// Existing header values are overwritten if collision happens.
-	Headers map[string]string `mapstructure:"headers"`
+	// Header values are opaque since they may be sensitive.
+	OpaqueHeaders map[string]configopaque.String `mapstructure:"headers"`
```

Comment:
Not convinced that I want to endup with this name. What do you think? Is there a way to use the previous name, maybe a few versions plan?

Reply:
My idea was to follow this plan: https://github.com/open-telemetry/opentelemetry-collector/pull/6470#issuecomment-1332094426 with which we would end up back with `Headers`

Reply:
I know that just doing `Headers map[string]configopaque.String` is a "backwards" incompatible change, but it is very limited incompatibility which I believe we can make an exception for to speed this up.

Reply:
I am fine with that, but I don't think @jpkrohling is, so I am trying to reach a compromise here :)

Reply:
Ping @jpkrohling

Reply:
I think the usage outside of contrib is minimal and the error and fix is quite clear (just change the type), so I change to do it in a single step. We can consider case by case on other fields